api/v1beta1/azuremachine_webhook.go

```diff
@@ -136,6 +136,20 @@ func (m *AzureMachine) ValidateUpdate(oldRaw runtime.Object) error {
 		)
 	}
 
+	if !reflect.DeepEqual(m.Spec.SubnetName, old.Spec.SubnetName) {
```

Comment:
Is this new immutability requirement for `SubnetName` closely related to this change? Should it be a separate PR?

Reply:
The immutability isn't strictly changed with these changes, however it appeared to be an oversight I believe while looking at managing and validating the new interface level field?